site/docs/blog/posts/2024-05-mariadb-mysql-support.md

```diff
@@ -0,0 +1,162 @@
+---
+date: 2024-05-24
+authors:
+  - adutra
+---
+
+# Support for MariaDB and MySQL
+
+We are happy to announce that Nessie now supports MariaDB and MySQL as backends. This is a
+significant milestone for Nessie, as it opens up new possibilities for Nessie users.
+
+A big thank you to [Vayuj Rajan](https://github.com/vyj7) for his contribution to this feature!
+
+<!-- more -->
+
+## Why MariaDB and MySQL?
+
+MariaDB and MySQL are popular open-source relational databases that are widely used in the industry.
+By adding support for MariaDB and MySQL, we are making it easier for users to get started with
+Nessie.
+
+**Important note**: MariaDB and MySQL support is considered experimental at this time, and we are
+looking for feedback from the community to improve it further.
+
+## How does it work?
+
+Nessie now ships with the MariaDB driver and the corresponding Quarkus extension. The MariaDB driver
+allows Nessie to work not only with MariaDB servers, but also with MySQL ones, thanks to MariaDB's
+protocol & schema compatibility with MySQL.
+
+## How to get started with MariaDB or MySQL
+
+To get started with MariaDB or MySQL, you can follow the instructions in the [Nessie
+documentation](https://projectnessie.org/nessie-latest/configuration/#jdbc-version-store-settings),
+but here is a quick step-by-step guide:
+
+1. Select the `JDBC` version store type as usual:
+
+```properties
+nessie.version.store.type=JDBC
+```
+
+2. Declare the JDBC datasource to use as `mariadb` (_this should be done regardless of whether you
+   are using MariaDB or MySQL_):
```

Comment:
Could we have `mysql` as an alias for user convenience? I imagine MySQL users will tend to think about their datasource as "MySQL" in daily activities... The JDBC URL will remain `jdbc:mariadb:`, of course, but config-wise it might be easier for users to manage ` quarkus.datasource.mysql.username=my_user`. WDYT?

Reply:
Interesting idea! Yes, I think we can add a 3rd datasource `quarkus.datasource.mysql`, its `db-kind` must be `mariadb` of course, but it could make the user XP slightly better indeed. Besides, if there are specificities for MySQL, these could easily go under `quarkus.datasource.mysql.jdbc.additional-jdbc-properties` without affecting MariaDB connections.

Also, it _might_ be possible to create a `ConfigSourceInterceptor` that would transparently change `jdbc:mysql` to `jdbc:mariadb` in all JDBC urls.